### ISSUE TITLE
When changing ApiVersionReader to MediaTypeApiVersionReader, swagger throws exception

### ISSUE THREAD

**Issue by ThomasCronholm** [Microsoft/aspnet-api-versioning#158]
When changing ApiVersionReader to MediaTypeApiVersionReader, swagger throws an exception.
This seems to be caused by the [FromBody] attribute.

I found this in my solution and reproduced it in your "SwaggerWebApiSample" project by changing in Startup.cs.

var httpServer = new HttpServer( configuration );

            // reporting api versions will return the headers "api-supported-versions" and "api-deprecated-versions"
            configuration.AddApiVersioning( o => {
                o.ReportApiVersions = true;
                o.ApiVersionReader = new MediaTypeApiVersionReader();
            } );
            configuration.MapHttpAttributeRoutes( constraintResolver );

This generates the following exception:
System.InvalidOperationException occurred
  HResult=0x80131509
  Message=The media type formatter JQueryMvcFormUrlEncodedFormatter could not be cloned. The type must either implement ICloneable or define a copy constructor with the signature 'JQueryMvcFormUrlEncodedFormatter(JQueryMvcFormUrlEncodedFormatter formatter)'.
  Source=Microsoft.AspNet.WebApi.Versioning.ApiExplorer
  StackTrace:
   at System.Web.Http.MediaTypeFormatterAdapterFactory.ResolveConstructor(Type type) in C:\z_Div\WebApiVersioning\aspnet-api-versioning-master003\aspnet-api-versioning-master\src\Microsoft.AspNet.WebApi.Versioning.ApiExplorer\System.Web.Http\MediaTypeFormatterAdapterFactory.cs:line 89
   at System.Web.Http.MediaTypeFormatterAdapterFactory.NewActivator(Type type) in C:\z_Div\WebApiVersioning\aspnet-api-versioning-master003\aspnet-api-versioning-master\src\Microsoft.AspNet.WebApi.Versioning.ApiExplorer\System.Web.Http\MediaTypeFormatterAdapterFactory.cs:line 54
   at System.Web.Http.MediaTypeFormatterAdapterFactory.NewCloneFunction(Type type) in C:\z_Div\WebApiVersioning\aspnet-api-versioning-master003\aspnet-api-versioning-master\src\Microsoft.AspNet.WebApi.Versioning.ApiExplorer\System.Web.Http\MediaTypeFormatterAdapterFactory.cs:line 45
   at System.Collections.Concurrent.ConcurrentDictionary`2.GetOrAdd(TKey key, Func`2 valueFactory)
   at System.Web.Http.MediaTypeFormatterAdapterFactory.GetOrCreateCloneFunction(MediaTypeFormatter formatter) in C:\z_Div\WebApiVersioning\aspnet-api-versioning-master003\aspnet-api-versioning-master\src\Microsoft.AspNet.WebApi.Versioning.ApiExplorer\System.Web.Http\MediaTypeFormatterAdapterFactory.cs:line 29
   at System.Web.Http.MediaTypeFormatterExtensions.Clone(MediaTypeFormatter formatter) in C:\z_Div\WebApiVersioning\aspnet-api-versioning-master003\aspnet-api-versioning-master\src\Microsoft.AspNet.WebApi.Versioning.ApiExplorer\System.Web.Http\MediaTypeFormatterExtensions.cs:line 13
   at Microsoft.Web.Http.Description.ApiVersionParameterDescriptionContext.CloneFormattersAndAddMediaTypeParameter(NameValueHeaderValue parameter, ICollection`1 formatters) in C:\z_Div\WebApiVersioning\aspnet-api-versioning-master003\aspnet-api-versioning-master\src\Microsoft.AspNet.WebApi.Versioning.ApiExplorer\Description\ApiVersionParameterDescriptionContext.cs:line 221
   at Microsoft.Web.Http.Description.ApiVersionParameterDescriptionContext.AddMediaTypeParameter(String name) in C:\z_Div\WebApiVersioning\aspnet-api-versioning-master003\aspnet-api-versioning-master\src\Microsoft.AspNet.WebApi.Versioning.ApiExplorer\Description\ApiVersionParameterDescriptionContext.cs:line 159
   at Microsoft.Web.Http.Description.ApiVersionParameterDescriptionContext.AddParameter(String name, ApiVersionParameterLocation location) in C:\z_Div\WebApiVersioning\aspnet-api-versioning-master003\aspnet-api-versioning-master\src\Microsoft.AspNet.WebApi.Versioning.ApiExplorer\Description\ApiVersionParameterDescriptionContext.cs:line 93
   at Microsoft.Web.Http.Versioning.MediaTypeApiVersionReader.AddParmeters(IApiVersionParameterDescriptionContext context) in C:\z_Div\WebApiVersioning\aspnet-api-versioning-master003\aspnet-api-versioning-master\src\Common\Versioning\MediaTypeApiVersionReader.cs:line 129
   at Microsoft.Web.Http.Description.VersionedApiExplorer.PopulateApiVersionParameters(ApiDescription apiDescription, ApiVersion apiVersion) in C:\z_Div\WebApiVersioning\aspnet-api-versioning-master003\aspnet-api-versioning-master\src\Microsoft.AspNet.WebApi.Versioning.ApiExplorer\Description\VersionedApiExplorer.cs:line 641
   at Microsoft.Web.Http.Description.VersionedApiExplorer.PopulateActionDescriptions(HttpActionDescriptor actionDescriptor, IHttpRoute route, String localPath, Collection`1 apiDescriptions, ApiVersion apiVersion) in C:\z_Div\WebApiVersioning\aspnet-api-versioning-master003\aspnet-api-versioning-master\src\Microsoft.AspNet.WebApi.Versioning.ApiExplorer\Description\VersionedApiExplorer.cs:line 777
   at Microsoft.Web.Http.Description.VersionedApiExplorer.ExploreDirectRouteControllers(HttpControllerDescriptor controllerDescriptor, IReadOnlyList`1 candidateActionDescriptors, IHttpRoute route, ApiVersion apiVersion) in C:\z_Div\WebApiVersioning\aspnet-api-versioning-master003\aspnet-api-versioning-master\src\Microsoft.AspNet.WebApi.Versioning.ApiExplorer\Description\VersionedApiExplorer.cs:line 568
   at Microsoft.Web.Http.Description.VersionedApiExplorer.InitializeApiDescriptions() in C:\z_Div\WebApiVersioning\aspnet-api-versioning-master003\aspnet-api-versioning-master\src\Microsoft.AspNet.WebApi.Versioning.ApiExplorer\Description\VersionedApiExplorer.cs:line 228
   at System.Lazy`1.CreateValue()

**Comment by commonsensesoftware** [comment]
This appears to be a duplicate of #156, where I provided a detailed explanation. The short, short answer is that behavior is by design in order to adequately support cloning (which is required). However, there are a handful of cases, such as yours, where the media type formatter might not contain a _copy constructor_. I agreed to also add support for parameterless constructors and the [v1.0.1](https://www.nuget.org/packages/Microsoft.AspNet.WebApi.Versioning.ApiExplorer/1.0.1) version of the package has been published containing this functionality. If you update your package, it should resolve your issue.

**Comment by ThomasCronholm** [comment]
Yes is fixes it and this is a duplicate.
Sorry for missing the new update release, and it being a duplicate.
Was looking through this last week, got interrupted and just returned to where i was today.
Thank you and keep up the good work :)

**Comment by commonsensesoftware** [comment]
No worries. It happens. Glad it unblocks you. :)
